pages/transcripts/[category]/[date]/[videoId]/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { parseISO } from "date-fns";
+import { getAllCategories, getAllVideosForCategory, getVideoForDateAndId, getSpeakerMapping, getTranscript, VideoData } from "../../../../../utilities/metadata-utils";
+import { formatDateForPath, parseDateFromPath } from "../../../../../utilities/path-utils";
+import { BoardMeeting } from '../../../../../components/BoardMeeting';
+import { GetStaticPaths, GetStaticProps } from "next";
+import { ReactElement } from "react";
+
+type VideoParams = {
+    category: string,
+    date: string,
+    videoId: string,
+}
+
+type VideoProps = {
+    metadata: any,
+    category: string,
+    transcript: any,
+    speakers: any
+};
+
+export const getStaticPaths = (async () => {
+    const categories: string[] = await getAllCategories();
+
+    const categoriesWithVideos: Array<{ category: string, video: VideoData }> = [];
+    
+    for (const category of categories) {
+        const videos: VideoData[] = await getAllVideosForCategory(category);
+
+        categoriesWithVideos.push(...videos.map(video => ({
+            category,
+            video,
+        })));
+    }
+
+    const nativeParams: Array<{ category: string, date: Date, videoId: string }> = categoriesWithVideos.map(({ category, video }) => ({
+        category: category,
+        date: video.date,
+        videoId: video.metadata.video_id
+    }));
+
+    return {
+        paths: nativeParams.map(entry => ({
+            params: {
+                category: entry.category,
+                date: formatDateForPath(entry.date),
+                videoId: entry.videoId
+            }
+        })),
+        fallback: false
+    };
+}) satisfies GetStaticPaths<VideoParams>;
+
+export const getStaticProps = (async (context) => {
+    const category = context.params.category;
+    const date = parseDateFromPath(context.params.date);
+    const videoId = context.params.videoId;
+
+    const video = await getVideoForDateAndId(category, date, videoId);
+
+    const transcript = await getTranscript(category, videoId);
+    const speakers = await getSpeakerMapping(category, videoId);
+
+    return {
+        props: {
+            metadata: video.metadata,
+            category,
+            transcript,
+            speakers
+        }
+    };
+}) satisfies GetStaticProps<VideoProps, VideoParams>;

```

Comment:
These might just be my typescript newbieness, but why does this have `VideoParams`?  I would have thought it was just `GetStaticProps<VideoProps>`?

Also, why the choice of `satisfies` instead of annotating the receiving variable?  Searching around, I found this example which is closer to what I would have expected:

https://www.vitamindev.com/next-js/getstaticprops-getstaticpaths-typescript/

Reply:
The second type parameter to `GetStaticProps` is the input params type, which allows you to ensure the contract between `getstaticPaths` and `getStaticProps` remains consistent.

For `satisfies` I was just using the template here: https://nextjs.org/docs/pages/building-your-application/configuring/typescript

I agree the linked way you linked seems simpler.